From 762e5b1131c0dae5770ec06c85e5eb2e4cb0072a Mon Sep 17 00:00:00 2001
From: Abel Vesa <abel.vesa@nxp.com>
Date: Mon, 29 Oct 2018 13:49:24 +0200
Subject: [PATCH 4959/5242] MLK-19966 clk: imx: Ignore gating unused composite
 clocks if m4 active

commit  4954072ba6b9d16a642d04fa9b8811551c3443e3 from
https://source.codeaurora.org/external/imx/linux-imx.git

Ignoring the gating of composite clocks if m4 is active is necessary
since any of those clocks can be in use by m4 at any given time.

Signed-off-by: Abel Vesa <abel.vesa@nxp.com>
Reviewed-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-composite-8m.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/imx/clk-composite-8m.c b/drivers/clk/imx/clk-composite-8m.c
index 27ab72d..076e5e2 100644
--- a/drivers/clk/imx/clk-composite-8m.c
+++ b/drivers/clk/imx/clk-composite-8m.c
@@ -166,6 +166,9 @@ struct clk *imx8m_clk_composite_flags(const char *name,
 	gate->reg = reg;
 	gate->bit_idx = PCG_CGC_SHIFT;
 
+	if (imx_src_is_m4_enabled())
+		flags |= CLK_IGNORE_UNUSED;
+
 	clk = clk_register_composite(NULL, name, parent_names, num_parents,
 					mux_hw, &clk_mux_ops, div_hw,
 					&imx8m_clk_composite_divider_ops,
-- 
1.7.9.5

